cmd/definitions/tmpl/service.tmpl

```diff
@@ -306,6 +303,17 @@ func (s *{{ $pn }})parsePair{{ $pn }}{{ $fnk }}(opts []Pair) (pair{{ $pn }}{{ $f
             err = s.formatError("{{$fn.Name}}", err {{ $path }} )
         }()
 
+        {{- if ne $fn.ObjectMode ""}}
+            {{- $mode := $fn.ObjectMode | toPascal }}
+            if !o.Mode.Is{{ $mode }}() {
+                err = services.ObjectModeInvalidError{Expected: Mode{{ $mode }}, Actual: o.Mode}
+                return
+            }
+        {{- end }}
+
+        pairs = append(pairs, s.defaultPairs.{{ $fnk }}...)
```

Comment:
Is there any reason to move the logic here?

Reply:
We should check and make sure `Object` is fit, otherwise we do not neet to handle pairs any more.

Reply:
LGTM